.github/workflows/update-emoji.yml

```diff
@@ -0,0 +1,53 @@
+name: Update Emoji
+
+on:
+  schedule:
+    - cron: '0 2 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: update-emoji-group
+
+jobs:
+  sync-emojis-from-github:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js v20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Fetch emojiData
+        run: npm run build:emoji
+
+      - name: Check for changes
+        run: |
+          if git diff --quiet -- "src/core/render/emoji-data.js"; then
+            echo "No changes Found. Skipping PR creation."
+          else
+            echo "Changes detected in emoji-data.js. Creating PR."
+
+            current_date=$(date +'%Y-%m-%d')
+
+            branch_name="update-emoji-data-$current_date"
+            pr_title="update(emoji): Sync emoji data with GitHub emoji API [$current_date]"
+            git config --global user.email "actions@github.com"
+            git config --global user.name "GitHub Actions"
+
+            git checkout -b $branch_name
+            git add src/core/render/emoji-data.js
+            git add docs/emoji.md
+            git commit -m "update(emoji): Sync emoji data with GitHub emoji API [$current_date]"
+            unique_branch_name=$branch_name-$(git log -1 --pretty=%h)
+            git checkout -b $unique_branch_name
+            git push origin $unique_branch_name
+            gh pr create --title "$pr_title" --body "Found updated github emojis need to sync, PTAL @docsifyjs/reviewers :robot:." --base develop
```

Comment:
This can be simplified by using the `action`

https://github.com/stefanzweifel/git-auto-commit-action
https://github.com/thomaseizinger/create-pull-request

Reply:
Thx for it.
I know those actions but according to our plain needs，using them gonna makes the action more complicated to me.
1. file change check steps and ::OUTPUT.
2. commit action on pre if-condition.
3. pr with change condition with if-condition also.

Personally, I think we could put them together without much actions dependencies for now.